pkg/operator/update/resources.go

```diff
@@ -0,0 +1,189 @@
+// TODO: Add unit test to check that various buried disk sizes get updated and
+// check that invalid cases (not matching) are left untouched. We can also check
+// that nothing besides the disk sizes is touched.
+func KeyspaceDiskSize(dst, src *planetscalev2.VitessKeyspaceTemplate) {
+	// Check that the keyspace definitions line up.
+	if !validateKeyspacePartitionings(dst.Partitionings, src.Partitionings) {
+		return
+	}
+
+	updateDiskSize(dst, src)
+}
+
+func updateDiskSize(dst, src *planetscalev2.VitessKeyspaceTemplate) {
+	for i := range dst.Partitionings {
+		dstPartitioning := &dst.Partitionings[i]
+		if dstPartitioning.Equal != nil {
+			updateEqualPartitioningDiskSize(dstPartitioning.Equal, src.Partitionings[i].Equal)
+		} else {
+			updateCustomPartitioningDiskSize(dstPartitioning.Custom, src.Partitionings[i].Custom)
+		}
+	}
+}
+
+func updateEqualPartitioningDiskSize(dst, src *planetscalev2.VitessKeyspaceEqualPartitioning) {
+	for i := range dst.ShardTemplate.TabletPools {
+		dstTablet := &dst.ShardTemplate.TabletPools[i]
+
+		requestedTablet := matchingTabletPool(dstTablet, src.ShardTemplate.TabletPools)
+		if requestedTablet == nil || requestedTablet.DataVolumeClaimTemplate == nil {
+			continue
+		}
+
+		dstRequests := &dstTablet.DataVolumeClaimTemplate.Resources.Requests
+		srcRequests := requestedTablet.DataVolumeClaimTemplate.Resources.Requests
+
+		// Apply the disk resource changes.
+		StorageResource(dstRequests, srcRequests)
+	}
+}
+
+func updateCustomPartitioningDiskSize(dst, src *planetscalev2.VitessKeyspaceCustomPartitioning) {
+	for i := range dst.Shards {
+		dstShard := &dst.Shards[i]
+		srcShard := &src.Shards[i]
+
+		for j := range dstShard.TabletPools {
+			dstTablet := &dstShard.TabletPools[j]
+
+			requestedTablet := matchingTabletPool(dstTablet, srcShard.TabletPools)
+			if requestedTablet == nil || requestedTablet.DataVolumeClaimTemplate == nil {
+				continue
+			}
+
+			dstRequests := &dstTablet.DataVolumeClaimTemplate.Resources.Requests
+			srcRequests := requestedTablet.DataVolumeClaimTemplate.Resources.Requests
+
+			// Apply the disk resource changes.
+			StorageResource(dstRequests, srcRequests)
+		}
+	}
+}
+
+func matchingTabletPool(dst *planetscalev2.VitessShardTabletPool, src []planetscalev2.VitessShardTabletPool) *planetscalev2.VitessShardTabletPool{
+	for i := range src {
+		srcTablet := &src[i]
+		if srcTablet.IsMatch(dst) {
+			return srcTablet
+		}
+	}
+
+	return nil
+}
+
+func validateKeyspacePartitionings(dst, src []planetscalev2.VitessKeyspacePartitioning) bool {
+	// Check that the list of partitionings are the same length.
+	if len(dst) != len(src) {
+		return false
+	}
+
+	for i := range dst {
+		// Validate that partitionings have the same type and shard count.
+		if !validatePartitionings(&dst[i], &src[i]) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func validatePartitionings(dst *planetscalev2.VitessKeyspacePartitioning, src *planetscalev2.VitessKeyspacePartitioning) bool {
```

Comment:
Same here. Functions that return bools should flow when used as conditional checks by callers:

```suggestion
func partitioningsMatch(dst *planetscalev2.VitessKeyspacePartitioning, src *planetscalev2.VitessKeyspacePartitioning) bool {
```

Reply:
Done.